runtime/include/sandbox_state_history.h

```diff
@@ -3,12 +3,30 @@
 #include "sandbox_state.h"
 #include "sandbox_types.h"
 
+#ifdef LOG_STATE_CHANGES
+#define SANDBOX_STATE_HISTORY_CAPACITY 100
+#else
+#define SANDBOX_STATE_HISTORY_CAPACITY 0
+#endif
+
+struct sandbox_state_history {
+	uint16_t        size;
+	sandbox_state_t buffer[SANDBOX_STATE_HISTORY_CAPACITY];
```

Comment:
Brute force approach to saving this state. I've never seen a sandbox have enough transitions to exhaust this buffer.